systems/data-plan/base-rate/cmd/server/main.go

```diff
@@ -3,74 +3,95 @@ package main
+	"github.com/ukama/ukama/systems/data-plan/base-rate/pkg/server"
+	"gopkg.in/yaml.v3"
 
-	"github.com/sirupsen/logrus"
 	log "github.com/sirupsen/logrus"
 	ccmd "github.com/ukama/ukama/systems/common/cmd"
 	ugrpc "github.com/ukama/ukama/systems/common/grpc"
+	mb "github.com/ukama/ukama/systems/common/msgBusServiceClient"
 	"github.com/ukama/ukama/systems/common/sql"
-	generated "github.com/ukama/ukama/systems/data-plan/base-rate/pb"
+	pb "github.com/ukama/ukama/systems/data-plan/base-rate/pb/gen"
 	"google.golang.org/grpc"
 )
 
-var serviceConfig *pkg.Config
+var serviceConfig = pkg.NewConfig(pkg.ServiceName)
 
 func main() {
 	ccmd.ProcessVersionArgument(pkg.ServiceName, os.Args, version.Version)
-	pkg.InstanceId = os.Getenv("POD_NAME")
+	log.Infof("Starting the base-rate service")
 
 	initConfig()
-	rateDb := initDb()
-	runGrpcServer(rateDb)
+	db := initDb()
+
+	runGrpcServer(db)
 }
 
-// initConfig reads in config file, ENV variables, and flags if set.
-func initConfig() {
-	serviceConfig = &pkg.Config{
-		DB: &uconf.Database{
-			DbName: pkg.ServiceName,
-		},
+func initDb() sql.Db {
+	log.Infof("Initializing Database")
+	d := sql.NewDb(serviceConfig.DB, serviceConfig.DebugMode)
+	err := d.Init(&db.Rate{})
+	if err != nil {
+		log.Fatalf("Database initialization failed. Error: %v", err)
 	}
+	return d
+}
+
+func initConfig() {
+	log.Infof("Initializing config")
+
 	err := config.NewConfReader(pkg.ServiceName).Read(serviceConfig)
 	if err != nil {
 		log.Fatal("Error reading config ", err)
 	} else if serviceConfig.DebugMode {
 		b, err := yaml.Marshal(serviceConfig)
 		if err != nil {
-			logrus.Infof("Config:\n%s", string(b))
+			log.Infof("Config:\n%s", string(b))
 		}
 	}
-
-	pkg.IsDebugMode = serviceConfig.DebugMode
 }
 
-func initDb() sql.Db {
-	log.Infof("Initializing Database")
-	d := sql.NewDb(serviceConfig.DB, serviceConfig.DebugMode)
-	err := d.Init(&db.Rate{})
-	if err != nil {
-		log.Fatalf("Database initialization failed. Error: %v", err)
+func runGrpcServer(d sql.Db) {
+
+	instanceId := os.Getenv("POD_NAME")
+	if instanceId == "" {
+		/* used on local machines */
+		inst := uuid.NewV4()
+		instanceId = inst.String()
 	}
-	return d
-}
 
-func runGrpcServer(gormdb sql.Db) {
-	grpcServer := ugrpc.NewGrpcServer(*serviceConfig.Grpc, func(s *grpc.Server) {
+	mbClient := mb.NewMsgBusClient(serviceConfig.MsgClient.Timeout, pkg.SystemName,
+		pkg.ServiceName, instanceId, serviceConfig.Queue.Uri,
+		serviceConfig.Service.Uri, serviceConfig.MsgClient.Host, serviceConfig.MsgClient.Exchange,
+		serviceConfig.MsgClient.ListenQueue, serviceConfig.MsgClient.PublishQueue,
+		serviceConfig.MsgClient.RetryCount,
+		serviceConfig.MsgClient.ListenerRoutes)
+
+	log.Debugf("MessageBus Client is %+v", mbClient)
 
-		srv := server.NewBaseRateServer(db.NewBaseRateRepo(gormdb))
-		generated.RegisterBaseRatesServiceServer(s, srv)
+	grpcServer := ugrpc.NewGrpcServer(*serviceConfig.Grpc, func(s *grpc.Server) {
```

Comment:
I am not really sure it's a good idea to have msgBus client without listener service. Hopefully it's not causing any issues. I think we can live without it for a while.

Reply:
It is not causing an issue